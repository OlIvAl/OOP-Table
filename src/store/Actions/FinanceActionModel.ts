import { IFinanceActionDetailData, IFinanceActionModel } from './interfaces';
import AbstractModelWithId from '../Abstract/AbstractModelWithId';
import { action, observable } from 'mobx';
import { IStore } from '../Abstract/interfaces';
import { actionType, statusType } from '../../@types/actions';

export class FinanceActionModel extends AbstractModelWithId<IStore<IFinanceActionModel>>
  implements IFinanceActionModel {
  @observable action_id: number = 0;
  @observable campaign_id: number = 0;
  @observable marker: string = '';
  @observable source: string = '';
  @observable action_state: statusType = 'processing';
  @observable created_at: string = '';
  @observable updated_at: string = '';
  @observable campaign_name: string = '';
  @observable price: number | null = null;
  @observable description: string = '';
  @observable profit_diff: number = 0;
  @observable total_profit: number = 0;
  @observable action_type?: actionType;

  @action.bound init(data: IFinanceActionDetailData) {
    this.action_id = data.action_id;
    this.campaign_id = data.campaign_id;
    this.marker = data.marker;
    this.source = data.source;
    this.action_state = data.action_state;
    this.created_at = data.created_at;
    this.campaign_name = data.campaign_name;
    this.price = data.price;
    this.total_profit = data.total_profit;
    this.action_type = data.action_type;
    this.updated_at = data.updated_at || '';
    this.description = data.description || '';
    this.profit_diff = data.profit_diff || 0;
  }
}
