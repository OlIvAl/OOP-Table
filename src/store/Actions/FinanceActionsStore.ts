import AbstractStoreWithCurrentModel from '../Abstract/AbstractStoreWithCurrentModel';
import { IFinanceActionModel } from './interfaces';

export class FinanceActionsStore extends AbstractStoreWithCurrentModel<IFinanceActionModel> {}
