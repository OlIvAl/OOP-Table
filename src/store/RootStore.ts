import { FinanceActionsStore } from './Actions/FinanceActionsStore';
import { IFinanceActionModel } from './Actions/interfaces';
import { IStoreWithCurrentModel } from './Abstract/interfaces';

export class RootStore {
  financeActionsStore: IStoreWithCurrentModel<IFinanceActionModel>;

  constructor() {
    this.financeActionsStore = new FinanceActionsStore(this);
  }
}