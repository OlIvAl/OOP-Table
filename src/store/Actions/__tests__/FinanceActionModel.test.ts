import { FinanceActionModel } from '../FinanceActionModel';
import { FinanceActionsStore } from '../FinanceActionsStore';
import { RootStore } from '../../RootStore';

describe('FinanceActionModel', () => {
  describe('constructor', () => {
    it('init valid model', () => {
      expect(new FinanceActionModel(new FinanceActionsStore(new RootStore()), 7)).toMatchSnapshot();
    });
  });
});
