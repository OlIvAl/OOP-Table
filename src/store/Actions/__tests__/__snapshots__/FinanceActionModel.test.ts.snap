// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FinanceActionModel constructor init valid model 1`] = `
FinanceActionModel {
  "id": 7,
  "store": FinanceActionsStore {
    "rootStore": RootStore {
      "financeActionsStore": FinanceActionsStore {
        "rootStore": [Circular],
      },
    },
  },
}
`;
